CHANGES.md

```diff
@@ -14,7 +14,24 @@
 [Unreleased](https://github.com/bird-house/birdhouse-deploy/tree/master) (latest)
 ------------------------------------------------------------------------------------------------------------------
 
-[//]: # (list changes here, using '-' for each new entry, remove this when items are added)
+## Changes
+- Jupyter: new image to add esgf-pyclient and xncml to fix Jenkins failure
+
+  See PR https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/118 for more details.
+
+  - Adds `esgf-pyclient` for esgf-dap.ipynb (https://github.com/Ouranosinc/pavics-sdi/pull/269)
+  - Adds `xncml` for gen_catalog refactoring (https://github.com/Ouranosinc/pavics-vdb/pull/46)
+  - Fixes annoying harmless error `ERROR 1: PROJ: proj_create_from_database: Open of /opt/conda/envs/birdy/share/proj failed`
+  - Relevant changes (alphabetical order):
```

Comment:
Hehe, let me quickly introduce you to our Jupyter env building process.

Incremental builds are builds on top of previously released Jupyter env.  DockerHub will use the `Dockerfile.testing` to build for incremental builds.  DockerHub knows which `Dockerfile` to use depending on the tag prefix.  For incremental build, the tag prefix is `dockerupdate-`.

In the `Dockerfile.testing` here https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/118/files#diff-b44bae84e9671e9e5f0a9f5a7f41cfb09c368daacc0873bab1cf3a57da8afdee you see I updated to the previous released version and I added `esgf-client` and `xncml`.  The rest of the changes (`gdal` and others), are pull indirectly.

If you look at the PR description, you see at the bottom there are [221130-221130-update230403-conda-env-export.diff.txt](https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/files/11151115/221130-221130-update230403-conda-env-export.diff.txt) which is the full diff vs the previous released Jupyter env.  You'll see more than just `gdal` but I only report `gdal` because it can potentially cause problem.

Full conda env export like [221130-update230403-conda-env-export.yml.txt](https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/files/11151116/221130-update230403-conda-env-export.yml.txt) are generated during each Jenkins builds so we know exactly what runtime env produce the error during the build.  See https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/blob/597529c35f0a9eabac92374b3797bff0f30509db/testall#L109-L114

The "Relevant changes" section is very "subjective".  It's what I considered "of interest" or "can cause trouble".

Full DockerHub builds will use regular `Dockerfile` and triggered by tag prefix `docker-`.

The name of our Jupyter env is funny too, it's `pavics/workflow-tests`.  Historically it was meant only for Jenkins to test the notebooks but then I realized I can deploy exactly the same image to production so I kill 2 birds with 1 stone.  I test the notebooks and I guaranty the runtime image works because both the notebooks and the runtime image are tested together.  I basically deploy exactly what I test.

UofT can have your own Jupyter env too.  CRIM actually have 2 more Jupyter env (EO and NLP at https://github.com/crim-ca/pavics-jupyter-images) that they deploy alongside our image for Climate.

The Jupyterhub can be configured to display several choices of Jupyter env image for use to choose, see https://github.com/bird-house/birdhouse-deploy/blob/60116e2c90c235b96df1b95dc55cd4b6d9458038/birdhouse/env.local.example#L232-L260

Reply:
Thank you for the detailed explanation!